File: dkb/convert_dkb_amount_test.go
package dkb

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestConvertDkbAmount_HappyPath(t *testing.T) {
	input := "55,94 €"

	output, _ := convertDkbAmount(input)
	expectedAmount := "55.94"

	assert.Equal(t, expectedAmount, output)
}

func TestConvertDkbAmount_SadPath(t *testing.T) {
	input := "asdf"

	_, err := convertDkbAmount(input)
	expectedError := "Failed to parse amount asdf"

	assert.Equal(t, expectedError, err.Error())
}

File: dkb/convertDkbAmount.test.ts
import { describe, it } from "@std/testing/bdd";
import { expect } from "@std/expect";
import { convertDkbAmount } from "./convertDkbAmount.ts";

describe("convertDkbAmount", () => {
  it("converts amount with currency symbol", () => {
    const input = "-55,94 €";
    const expectedAmount = "-55.94";
    expect(convertDkbAmount(input)).toBe(expectedAmount);
  });

  it("converts amount without currency symbol", () => {
    const input = "-55,94";
    const expectedAmount = "-55.94";
    expect(convertDkbAmount(input)).toBe(expectedAmount);
  });

  it("throws error for invalid input", () => {
    const input = "asdf";
    expect(() => convertDkbAmount(input)).toThrow("Failed to parse amount asdf");
  });
});
